### PR TITLE
feat: add support for custom expect message

### LINKED ISSUE #1310: [Feature] Add custom expect messages
I was looking for a way to specify custom error messages for assertions and it looks like they are available in [Node.js](https://playwright.dev/docs/test-assertions#custom-expect-message) Playwright. How much effort would it be to also implement them in Python? With a little hint on where to start, I could give it a try.

### PR DESCRIPTION
**Pull request by sawt00th**
RP add support custom message to expect expressions and resolve #1310 issue

### CODE REVIEW

**Comment by sawt00th** [comment]
@microsoft-github-policy-service agree [company="default"]


ср, 25 янв. 2023 г. в 22:43, microsoft-github-policy-service[bot] <
***@***.***>:

**Comment by sawt00th** [comment]
@microsoft-github-policy-service agree


чт, 26 янв. 2023 г. в 00:35, microsoft-github-policy-service[bot] <
***@***.***>:

**Comment by sawt00th** [comment]
@mxschmitt can you review this one?

**Comment by sawt00th** [comment]
@mxschmitt do i need some fixes?

File: tests/async/test_assertions.py [hunk 1]
```diff
@@ -658,3 +658,16 @@ async def test_should_print_response_with_text_content_type_if_to_be_ok_fails(
...
     assert "Image content type error" not in error_message
+
+
+async def test_should_print_users_message_for_page_based_assertion(
+    page: Page, server: Server
+) -> None:
+    await page.goto(server.EMPTY_PAGE)
+    await page.set_content("<title>new title</title>")
+    with pytest.raises(AssertionError) as excinfo:
+        await expect(page, "Title is not new").to_have_title("old title")
```
Comment (mxschmitt) [inline tests/async/test_assertions.py:669]:
```suggestion
        await expect(page, "Title is not new").to_have_title("old title", timeout=100)
```

this makes the test pass faster.

File: tests/async/test_assertions.py [hunk 2]
```diff
@@ -658,3 +658,16 @@ async def test_should_print_response_with_text_content_type_if_to_be_ok_fails(
...
+    assert "Title is not new" in str(excinfo.value)
+    with pytest.raises(AssertionError) as excinfo:
+        await expect(page).to_have_title("old title")
```
Comment (mxschmitt) [inline tests/async/test_assertions.py:672]:
```suggestion
        await expect(page).to_have_title("old title", timeout=100)
```

File: tests/sync/test_assertions.py [hunk 3]
```diff
@@ -746,3 +747,16 @@ def test_should_print_response_with_text_content_type_if_to_be_ok_fails(
...
     assert "Image content type error" not in error_message
+
+
+def test_should_print_users_message_for_page_based_assertion(
+    page: Page, server: Server
+) -> None:
+    page.goto(server.EMPTY_PAGE)
+    page.set_content("<title>new title</title>")
+    with pytest.raises(AssertionError) as excinfo:
+        expect(page, "Title is not new").to_have_title("old title")
```
Comment (mxschmitt) [inline tests/sync/test_assertions.py:758]:
```suggestion
        expect(page, "Title is not new").to_have_title("old title", timeout=100)
```

File: tests/sync/test_assertions.py [hunk 4]
```diff
@@ -746,3 +747,16 @@ def test_should_print_response_with_text_content_type_if_to_be_ok_fails(
...
+    assert "Title is not new" in str(excinfo.value)
+    with pytest.raises(AssertionError) as excinfo:
+        expect(page).to_have_title("old title")
```
Comment (mxschmitt) [inline tests/sync/test_assertions.py:761]:
```suggestion
        expect(page).to_have_title("old title", timeout=100)
```

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**